### PR TITLE
[Mono.Android] Bind Android API-36 DP1.

### PR DESCRIPTION
**Pull request by jpobst**
Context: https://developer.android.com/about/versions/16
Context: https://android-developers.googleblog.com/2024/11/the-first-developer-preview-android16.html

Android 16 Developer Preview 1 has been released.  The Android 16
Developer Preview Program Overview [Timeline and updates][0] section
suggests the following timeline:

  * Nov/Dec: Developer Previews
  * Jan/Feb: Unstable Betas
  * Mar/Apr: Stable Betas
  * ???: Final

Currently, this will be usable in its preview form to `main` users who explicitly target `net9.0-android36`.  Once we are shipping .NET 10 previews, it will be usable for users who explicitly target `net10.0-android36`.  We will need to decide on our strategy for backporting this to .NET 9 service releases.

`PublicApiAnalyzers` reports 2 categories of API changes (`PublicAPI.Shipped.txt`):

- NRT attribute changes (`?` -> `!`)
- Method parameter name changes, technically a source breaking change, but something we've never tracked or fixed before

[0]: https://developer.android.com/about/versions/16/overview

### CODE REVIEW

File: build-tools/scripts/DotNet.targets [hunk 1]
```diff
@@ -29,18 +29,22 @@
...
     <MSBuild
          Condition=" '$(AndroidDefaultTargetDotnetApiLevel)' != '$(AndroidLatestStableApiLevel)' "
          Projects="$(_Root)src\Mono.Android\Mono.Android.csproj"
-         Properties="TargetFramework=$(DotNetTargetFramework);AndroidApiLevel=$(AndroidDefaultTargetDotnetApiLevel);AndroidPlatformId=$(AndroidDefaultTargetDotnetApiLevel);DisableApiCompatibilityCheck=true"
+         Properties="AndroidApiLevel=$(AndroidDefaultTargetDotnetApiLevel);AndroidPlatformId=$(AndroidDefaultTargetDotnetApiLevel)"
     />
     <MSBuild
          Condition=" '$(AndroidLatestUnstableApiLevel)' != '$(AndroidLatestStableApiLevel)' "
          Projects="$(_Root)src\Mono.Android\Mono.Android.csproj"
-         Properties="TargetFramework=$(DotNetTargetFramework);AndroidApiLevel=$(AndroidLatestUnstableApiLevel);AndroidPlatformId=$(AndroidLatestUnstablePlatformId);DisableApiCompatibilityCheck=true"
```
Comment (pjcollins) [inline build-tools/scripts/DotNet.targets:42]: We can add these DisableApiCompatibilityCheck=true properties back when building on Windows if necessary, but wanted to test again without them first

**Comment by pjcollins** [comment]
@jpobst I think this is building correctly now, but there may be some ApiCompat things to look at

File: src/Mono.Android/metadata [hunk 2]
```diff
@@ -1953,6 +1953,12 @@
...
   <attr api-since="35" path="/api/package[@name='android.service.notification']/class[@name='Condition']/constructor[@name='Condition' and count(parameter)=8 and parameter[1][@type='android.net.Uri'] and parameter[2][@type='java.lang.String'] and parameter[3][@type='java.lang.String'] and parameter[4][@type='java.lang.String'] and parameter[5][@type='int'] and parameter[6][@type='int'] and parameter[7][@type='int'] and parameter[8][@type='int']]/parameter[7]" name="enumType">Android.Service.Notification.ConditionSource</attr>
 
+  <!-- Rename Namespaces -->
+  <attr api-since="36" path="/api/package[@name='android.app.appfunctions']" name="managedName">Android.App.AppFunctions</attr>
+  <attr api-since="36" path="/api/package[@name='android.app.appsearch.functions']" name="managedName">Android.App.AppSearch.Functions</attr>
+  <attr api-since="36" path="/api/package[@name='android.configinfrastructure.aconfig']" name="managedName">Android.ConfigInfrastructure.Aconfig</attr>
```
Comment (jonathanpeppers) [inline src/Mono.Android/metadata:1959]: I don't know if we have a past example, but if `aconfig` is short for "Android Config", should the C# namespace be `Android.ConfigInfrastructure.AConfig`?

**Review by jonathanpeppers — APPROVED** [review]
Looks reasonable to me. 👍

File: src/Mono.Android/metadata (hunk 2 above)
Comment (jpobst) [reply src/Mono.Android/metadata:1959]: Yeah, I researched this one to see what Google's intent is.  I generally defer to the way they have named their types which in this case is indeed `Aconfig`:

https://developer.android.com/reference/android/configinfrastructure/aconfig/AconfigPackage?hl=en

**Comment by jonpryor** [comment]
@jpobst: should this also update `$(AndroidLatestUnstableApiLevel)`/etc.?

https://github.com/dotnet/android/blob/894ecea3effa6e96ab7545c60ffcd45c819b2d52/Configuration.props#L32-L34

This would make it "reasonable" for me to add `--preview-platform-version to `WorkloadDependencies.json` in #9613.

**Comment by jonpryor** [comment]
@jonpryor [wrote](https://github.com/dotnet/android/pull/9594#issuecomment-2541863840):

> should this also update `$(AndroidLatestUnstableApiLevel)`/etc.?

It did, I just didn't see it.  Doh!

**Pull request merged**